### PR TITLE
chore: add MintManager deploy script

### PR DESCRIPTION
**Pull request by Pangssu**
## Summary by CodeRabbit

- **New Features**
    - Enhanced minting capabilities with the introduction of the `MintManager` contract deployment on Layer 2.
    - Expanded blockchain connectivity by adding network configurations for `holesky` and `kromaHolesky`.
    - Added configurations for the Ethereum Holesky testnet, including RPC settings and a private key for the deployer account.
    - Introduced companion networks for cross-network referencing.

### CODE REVIEW

**Review by 0xHansLee — APPROVED** [review]

File: packages/contracts/deploy/L2/012-MintManager.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      )
+      await assertContractVariable(
+        contract,
+        'SLIDING_WINDOW_BLOCKS',
+        deployConfig.mintManagerSlidingWindowBlocks
+      )
+      await assertContractVariable(
+        contract,
+        'DECAYING_FACTOR',
+        deployConfig.mintManagerDecayingFactor
```
Comment (seolaoh) [inline packages/contracts/deploy/L2/012-MintManager.ts:36]: Please add check `GOVERNANCE_TOKEN` is predeploy address of governance token.

**Review by sm-stack — APPROVED** [review]

File: packages/contracts/hardhat.config.ts [hunk 2]
```diff
@@ -46,6 +46,27 @@ const config: HardhatUserConfig = {
       },
       deploy: ['./deploy/L2'],
     },
+    holesky: {
+      chainId: 17000,
+      url: process.env.L1_RPC_HOLESKY || '',
+      accounts: [
+        process.env.PRIVATE_KEY_DEPLOYER_HOLESKY || ethers.constants.HashZero,
```
Comment (seolaoh) [inline packages/contracts/hardhat.config.ts:53]: Why don't separate `PRIVATE_KEY_DEPLOYER_HOLESKY`?

File: packages/contracts/hardhat.config.ts [hunk 3]
```diff
@@ -46,6 +46,27 @@ const config: HardhatUserConfig = {
...
+      ],
+      companionNetworks: {
+        l1: 'kromaHolesky',
+      },
+      deploy: ['./deploy/L1'],
+    },
+    kromaHolesky: {
+      url: process.env.L2_RPC_KROMA_HOLESKY || '',
```
Comment (seolaoh) [inline packages/contracts/hardhat.config.ts:61]: `chainId` is missing.

**Review by seolaoh — APPROVED** [review]
LGTM

File: packages/contracts/.env.example [hunk 4]
```diff
@@ -10,8 +10,13 @@ L2_RPC_KROMA_MAINNET=
...
+# RPC for Ethereum holesky testnet
+L1_RPC_HOLESKY=
+# RPC for Kroma sepolia testnet
+L2_RPC_KROMA_HOLESKY=
 
 # Private key for the deployer account
 PRIVATE_KEY_DEPLOYER=
 PRIVATE_KEY_DEPLOYER_MAINNET=
 PRIVATE_KEY_DEPLOYER_SEPOLIA=
+PRIVATE_KEY_DEPLOYER_HOLESKY=
```
[the remark this answers is not in the document]
Comment (Pangssu) [reply packages/contracts/.env.example:22]: This is a sample of .env file

**Pull request merged**